pczt/src/common.rs

```diff
@@ -0,0 +1,44 @@
+/// Global fields that are relevant to the transaction as a whole.
+#[derive(Clone)]
+pub(crate) struct Global {
+    //
+    // Transaction effecting data.
+    //
+    // These are required fields that are part of the final transaction, and are filled in
+    // by the Creator when initializing the PCZT.
+    //
+    pub(crate) tx_version: u32,
+    pub(crate) version_group_id: u32,
```

Comment:
Should this pair of fields be optional independently of the others?

Reply:
Would it be better to infer the version group id from the tx version instead of having it as an explicit field?

> Should this pair of fields be optional independently of the others?

Would it make sense to adding a builder struct where the `tx_version` field is optional instead of making it optional here?

Reply:
> Would it be better to infer the version group id from the tx version instead of having it as an explicit field?

That's more fragile across consensus forks, but I can see an argument that the tuple `(tx_version, consensus_branch_id)` might be sufficient to fully constrain the value of `version_group_id` such that every role can reliably deduce it. The main argument against this would be that it would force PCZT roles to be aware of more details about transaction formats, where they might otherwise just be able to focus on the PCZT.

Reply:
IMHO the arguments for encoding the version group ID explicitly in the transaction format apply equally to PCZTs.